src/microbe_stage/systems/CompoundAbsorptionSystem.cs

```diff
@@ -55,6 +55,13 @@ protected override void Update(float delta, in Entity entity)
             throw new NotImplementedException();
         }
 
+        if (entity.Has<MicrobeControl>())
+        {
+            // Microbe can't passively absorb compounds with mucocyst active
+            if (entity.Get<MicrobeControl>().State == MicrobeState.MucocystShield)
```

Comment:
It would be a ton cleaner if enabling mucocyst shield instead set the absorb speed to 0 and restored it later, that way there's less intertangling of the systems when absorption system won't need to know that mucocyst feature exists.